### PR TITLE
DOC/Gallery example "Cross-section along a transect": General improvements

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR contains general improvements of the Gallery example "Cross-section along a transect":
-  Use lists as input for the `center` and `endpoint` parameters
- Make argument passed to `xshift` flexible
- Re-wrapp to 88 chars
- Improve comments

**Preview**: https://pygmt-dev--3623.org.readthedocs.build/en/3623/gallery/images/cross_section.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

**Comment by seisman** [comment]
The changes look good to me. Could you please also rewrap the lines to <88 characters?

**Comment by yvonnefroehlich** [comment]
> The changes look good to me. Could you please also rewrap the lines to <88 characters?

Done in https://github.com/GenericMappingTools/pygmt/pull/3623/commits/afbbdab36dcd2c5ec2ae4dbd79025ac5437f166c.

File: examples/gallery/images/cross_section.py [hunk 1]
```diff
@@ -2,13 +2,11 @@
...
-In this example, the elevation is extracted from a grid provided via
-:func:`pygmt.datasets.load_earth_relief`.
-The figure consists of two parts, a map of the elevation in the study
-area showing the survey line and a Cartesian plot showing the elevation
-along the survey line.
+:func:`pygmt.project` and :func:`pygmt.grdtrack` can be used to focus on a quantity
+and its variation along a desired survey line. In this example, the elevation is
+extracted from a grid provided via :func:`pygmt.datasets.load_earth_relief`. The
+figure consists of two parts, a map of the elevation in the study area showing the
+survey line and a Cartesian plot showing the elevation along the survey line.
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:9]:
```suggestion
:func:`pygmt.project` and :func:`pygmt.grdtrack` can be used to focus on a quantity and 
its variation along a desired survey line. In this example, the elevation is extracted 
from a grid provided via :func:`pygmt.datasets.load_earth_relief`. The figure consists 
of two parts, a map of the elevation in the study area showing the survey line and a 
Cartesian plot showing the elevation along the survey line.
```

File: examples/gallery/images/cross_section.py [hunk 2]
```diff
@@ -34,8 +32,8 @@
     frame="af",
 )
 
-# Download grid for Earth relief with a resolution of 10 arc-minutes and
-# gridline registration [Default]
+# Download grid for Earth relief with a resolution of 10 arc-minutes and gridline
+# registration [Default]
 grid_map = pygmt.datasets.load_earth_relief(
     resolution="10m",
     region=region_map,
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:39]:
```suggestion
    region=region_map
```

File: examples/gallery/images/cross_section.py [hunk 3]
```diff
@@ -79,16 +75,16 @@
...
-fig.shift_origin(yshift="12.5c")
+# Shift plot origin to the top by the hight of the map ("+h") plus 1.5 centimeters
+fig.shift_origin(yshift="h+1.5c")
 
 fig.basemap(
     region=[0, 15, -8000, 6000],  # x_min, x_max, y_min, y_max
-    # Cartesian projection with a width of 12 centimeters and
-    # a height of 3 centimeters
+    # Cartesian projection with a width of 12 centimeters and a height of 3
+    # centimeters
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:84]:
```suggestion
    # Cartesian projection with a width of 12 centimeters and a height of 3 centimeters
```

File: examples/gallery/images/cross_section.py [hunk 4]
```diff
@@ -101,16 +97,16 @@
...
     generate="0.1",  # Output data in steps of 0.1 degrees
 )
 
-# Extract the elevation at the generated points from the downloaded grid
-# and add it as new column "elevation" to the pandas.DataFrame
+# Extract the elevation at the generated points from the downloaded grid and add
+# it as new column "elevation" to the pandas.DataFrame
 track_df = pygmt.grdtrack(
     grid=grid_map,
     points=track_df,
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:112]: Line 117, remove the trailing comma, and run `make format`.

File: examples/gallery/images/cross_section.py [hunk 5]
```diff
@@ -101,28 +89,24 @@
...
-# Generate points along a great circle corresponding to the survey line
-# and store them in a pandas.DataFrame
+# Generate points along a great circle corresponding to the survey line and store
+# them in a pandas.DataFrame
 track_df = pygmt.project(
-    center="126/42",  # Start point of survey line (longitude/latitude)
-    endpoint="146/40",  # End point of survey line (longitude/latitude)
+    center=[126, 42],  # Start point of survey line (longitude, latitude)
+    endpoint=[146, 40],  # End point of survey line (longitude, latitude)
     generate="0.1",  # Output data in steps of 0.1 degrees
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:97]:
```suggestion
    generate=0.1,  # Output data in steps of 0.1 degrees
```

File: examples/gallery/images/cross_section.py [hunk 6]
```diff
@@ -101,28 +89,24 @@
...
 
-# Extract the elevation at the generated points from the downloaded grid
-# and add it as new column "elevation" to the pandas.DataFrame
-track_df = pygmt.grdtrack(
-    grid=grid_map,
-    points=track_df,
-    newcolname="elevation",
-)
+# Extract the elevation at the generated points from the downloaded grid and add
+# it as new column "elevation" to the pandas.DataFrame
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:101]:
```suggestion
# Extract the elevation at the generated points from the downloaded grid and add it as 
# new column "elevation" to the pandas.DataFrame
```

File: examples/gallery/images/cross_section.py [hunk 7]
```diff
@@ -101,28 +89,24 @@
     font="10p",  # Use a font size of 10 points
 )
 
-# Generate points along a great circle corresponding to the survey line
-# and store them in a pandas.DataFrame
+# Generate points along a great circle corresponding to the survey line and store
+# them in a pandas.DataFrame
```
Comment (seisman) [inline examples/gallery/images/cross_section.py:93]:
```suggestion
# Generate points along a great circle corresponding to the survey line and store them 
# in a pandas.DataFrame
```

File: examples/gallery/images/cross_section.py [hunk 8]
```diff
@@ -79,16 +68,15 @@
 # ----------------------------------------------------------------------------
 # Top: Elevation along survey line
 
-# Shift plot origin 12.5 centimeters to the top
-fig.shift_origin(yshift="12.5c")
+# Shift plot origin to the top by the hight of the map ("+h") plus 1.5 centimeters
```
Comment (michaelgrund) [inline examples/gallery/images/cross_section.py:71]:
```suggestion
# Shift plot origin to the top by the height of the map ("+h") plus 1.5 centimeters
```

**Review by seisman — APPROVED** [review]

**Pull request merged**